docs/architecture/adr-069-gov-improvements.md

```diff
@@ -0,0 +1,198 @@
+  string metadata = 3;
+  string title = 4;
+  string summary = 5;
+  string option_one = 6;
+  string option_two = 7;
+  string option_three = 8;
+  string option_four = 9;
+}
+```
+
+Voters can only vote on the defined options in the proposal.
+
+To maintain compatibility with the existing endpoints, the voting options will not be stored in the proposal itself and each option will be mapped to [`v1.VoteOption`][5]. A multiple choice proposal will be stored as a [`v1.Proposal`][5]. A query will be available for multiple choice proposal types to get the voting options.
+
+### Votes
+
+As mentioned above [multiple choice proposal](#multiple-choice-proposal) will introduce an additional vote option: `SPAM`.
+
+This vote option will be supported by all proposal types.
+At the end of the voting period, if a proposal is voted as `SPAM`, its fails and its deposit is burned.
+
+`SPAM` differs from the `No with Veto` vote as its threshold is dynamic.
+A proposal is marked as `SPAM` when the total of weighted votes for all options is lower than the amount of weighted vote on `SPAM`
+(`spam` > `option_one + option_two + option_three + option_four` = proposal marked as spam).
+
+To avoid voters wrongfully voting down a proposal as `SPAM`, voters will be slashed `x`% (default 0%) of their voting stake if they voted `SPAM` on a proposal that wasn't a spam proposal. The parameter allows to incentivise voters to only vote `SPAM` on actual spam proposals and not use `SPAM` as a way to vote `No with Veto` with a different threshold.
+
+This leads to the addition of the following governance parameter in [`v1.Params`][5]:
+
+```protobuf
+// burn_spam_amount defines the percentage of the voting stake that will be burned if a voter votes SPAM on a proposal that is not marked as SPAM.
+string burn_spam_amount = 8 [(cosmos_proto.scalar) = "cosmos.Dec"];
+```
+
+Additionally, the current vote options will be renamed to better accommodate the multiple choice proposal:
+
+```protobuf
+// VoteOption enumerates the valid vote options for a given governance proposal.
+enum VoteOption {
+  VOTE_OPTION_UNSPECIFIED = 0;
+  VOTE_OPTION_ONE = 1;
+  VOTE_OPTION_TWO = 2;
+  VOTE_OPTION_THREE = 3;
+  VOTE_OPTION_FOUR = 4;
+  VOTE_OPTION_SPAM = 5;
+}
+```
+
+The order does not change for a normal proposal (1 = yes, 2 = abstain, 3 = no, 4 = no with veto as it was). We will attempt to make this change in a non-breaking way for clients (cli and wallets).
+
+Updating vote options means updating [`v1.TallyResult`][5] as well.
+
+#### Tally
+
+Due to the vote option change, each proposal can have the same tallying method.
+
+However, chains may want to change the tallying function (weighted vote per voting power) of `x/gov` for a different algorithm (using a quadratic function on the voter stake, for instance).
+
+The custom tallying function can be passed to the `x/gov` keeper with the following interface:
+
+```go
+type Tally interface{
```

Comment:
I need to think a bit more of what exactly we need but at a high level you should be able to process a custom result with the voting power that has voted so far

Reply:
> at a high level you should be able to process a custom result with the voting power that has voted so far

could you elaborate here? I dont understand what you mean